napari/layers/base/_base_mouse_bindings.py

```diff
@@ -51,7 +50,9 @@ def _translate_with_box(
 ) -> None:
     offset = mouse_pos - initial_mouse_pos
     new_affine = Affine(translate=offset).compose(initial_affine)
-    layer.affine = layer.affine.replace_slice(event.dims_displayed, new_affine)
+    layer.affine = layer.affine.replace_slice(
+        layer._slice_input.displayed, new_affine
+    )
```

Comment:
So event is no longer used?

Reply:
Correct, the event `dims_displayed` info is only used for the position. Just in case, a more detailed description of this approach is over https://github.com/napari/napari/issues/7333#issuecomment-2433259437. However, if there is a different way to make things work let me know! For example, checking again things here, I stumbled across a helper function called `dims_displayed_world_to_layer`. Maybe it could be better to use that instead of directly using `layer._slice_input.displayed`? 🤔

Reply:
Just in case, the helper function is https://github.com/napari/napari/blob/b9dc8dca1e1afa7567586e7ffa7a36cc7cd9fb6f/napari/layers/utils/layer_utils.py#L651-L669